### PR TITLE
feat: add renderActiveBullet

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
This adds `renderActiveBulelt` option for the plugin `Pagination`.
- The default value is `null`
- If `null`, the active bullet is rendered with `renderBullet` with `bullet-active` class, same as before.
- If a render function is given, the active bullet is rendered with that function.

## Why is this needed?
The proposed use case is this:
- The pagination should be rendered with a page index inside
  - Like `<span>1</span>`, `<span>2</span>` and so on
- The active pagination bullet should have different text inside.
  - Like `<span>ACTIVE</span>`

A different approach for this(which works well too) is by using CSS `content` and `counter`.
- https://codepen.io/woodneck/pen/RwBjZRE

But, for a11n I had to add this feature.

### CODE REVIEW

File: src/pagination/renderer/ScrollBulletRenderer.ts [hunk 1]
```diff
@@ -95,6 +90,29 @@ class ScrollBulletRenderer extends Renderer {
...
 
+    if (renderActiveBullet) {
+      const prevBullet = bullets[prevIndex];
+      if (prevBullet) {
+        const newBullet = this._createBulletFromString(
+          renderBullet(bulletClass, prevIndex),
+          prevIndex
+        );
+        prevBullet.parentElement!.replaceChild(newBullet, prevBullet);
+        bullets.splice(prevIndex, 1, newBullet);
```
Comment (daybrush) [inline src/pagination/renderer/ScrollBulletRenderer.ts:101]: It seems that you can use it as follows without removing or creating an index.
```
bullets[prevIndex] = nextBullet;
```

File: src/pagination/renderer/ScrollBulletRenderer.ts [hunk 2]
```diff
@@ -95,6 +90,29 @@ class ScrollBulletRenderer extends Renderer {
     const prevClassRegex = new RegExp(`^${prevClassPrefix}`);
     const nextClassRegex = new RegExp(`^${nextClassPrefix}`);
 
+    if (renderActiveBullet) {
+      const prevBullet = bullets[prevIndex];
+      if (prevBullet) {
+        const newBullet = this._createBulletFromString(
```
Comment (daybrush) [inline src/pagination/renderer/ScrollBulletRenderer.ts:96]: There seems to be a task of creating an element every time it changes. How about caching?

**Review by daybrush — APPROVED** [review]
LGTM

File: src/pagination/renderer/ScrollBulletRenderer.ts (hunk 2 above)
Comment (WoodNeck) [reply src/pagination/renderer/ScrollBulletRenderer.ts:96]: Well, that would be good if the rendered result is consistent.
As `renderActiveBullet` takes two parameters, `className` and `index`, I assumed that every render result can differ, and also the options can change at any time so I made it like that. I agree that it's not the best way of doing it.

File: src/pagination/renderer/ScrollBulletRenderer.ts (hunk 1 above)
Comment (WoodNeck) [reply src/pagination/renderer/ScrollBulletRenderer.ts:101]: That would be better in this case 👍

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**